website/blog/corba.md

```diff
@@ -0,0 +1,184 @@
+```
+
+With Service Weaver, remote method calls are syntactically identical to local
+method calls. They do **look** the same, and that's okay.
+
+```go
+// A remote method call in Service Weaver.
+sum, err := adder.Add(ctx, 1, 2)
+```
+
+## Acting Local
+
+When we say a remote method call **acts** like a local method call, we mean they
+have the same **semantics**. This is where frameworks can get into trouble.
+[The fallacies of distributed computing][fallacies] explain that remote method
+calls have fundamentally different semantics than local method calls. For
+example:
+
+- Remote method calls exhibit partial (and sometimes undetectable) failures.
+  Local method calls can leverage fate sharing; if something crashes, everything
+  crashes.
+- Remote method calls execute with significantly higher latency than local
+  method calls.
+- Remote method call arguments and results must be copied, but local method
+  calls can use references.
+
+Because of these fundamental semantic differences, trying to make a remote
+method call act like local method call will not end well. Take distributed
+shared memory, for example. The idea behind distributed shared memory is that a
+regular single-machine program can be run, without modification, across multiple
+machines. The program's low-level memory loads and stores are translated to be
+performed against remote memory. Distributed shared memory never took off
+because hiding the latency and partial failures of remote memory accesses is
+extremely challenging.
+
+Or consider NFS hard mounts, an example taken from [*A Note on Distributed
+Computing*][a_note_on_dc]. If an operation on a hard mounted NFS volume fails,
+the NFS client tries to mask the failure by retrying the operation indefinitely
+until it succeeds. This means that if any NFS server fails, all clients
+freeze until it is repaired.
+
+While Service Weaver remote method calls look like local method calls, they do
+**not** act like them.
+
+- Remote methods return an error on network and participant failures.  The
+  developer is responsible for handling such errors.
+- Every remote method's first argument is a `context.Context`. The
+  developer is responsible for setting deadlines on these contexts to detect and
+  react to method calls that are experiencing high latency.
+- The developer is responsible for making all arguments and returns of a remote
+  method call serializable, which is enforced by the [Service Weaver code
+  generator][generator].
+
+Yes, these are a burden on the programmer, but that is precisely the point of
+[*A Note on Distributed Computing*][a_note_on_dc]; it is a mistake to try to
+isolate the programmer from the fundamental issues of remote calls. We are also
+working on extending our [unit test framework][testing] to automatically inject
+latency and failures to further ensure that Service Weaver applications are
+resilient to the failures and delays they will encounter in a real deployment.
+
```

Comment:
Might be a good place to put in the "networked disk" addendum here, espousing the speed and the resilience of modern cloud networks. WDYT?